src/main/java/com/meilisearch/sdk/Client.java

```diff
@@ -242,4 +248,58 @@ public Key createKey(Key options) throws Exception {
     public void deleteKey(String key) throws Exception {
         this.keysHandler.deleteKey(key);
     }
+
+    public String generateTenantToken(Map<String, Object> searchRules) throws MeiliSearchException {
+        return this.generateTenantToken(searchRules, new TenantTokenOptions());
+    }
+
+    /**
+     * Generate a tenant token
+     *
+     * @param searchRules A Map of string, object which contains the rules to be enforced at search
+     *     time for all or specific accessible indexes for the signing API Key.
+     * @param options A TenantTokenOptions, the following fileds are accepted: - apiKey: String
+     *     containing the API key parent of the token. If you leave it empty the client API Key will
+     *     be used. - expiresAt: A DateTime when the key will expire. Note that if an expiresAt
+     *     value is included it should be in UTC time.
+     * @return String containing the tenant token
+     * @throws MeiliSearchException if an error occurs
+     */
+    public String generateTenantToken(Map<String, Object> searchRules, TenantTokenOptions options)
+            throws MeiliSearchException {
+        // Validate all fields
+        if ((options.getApiKey() == null || options.getApiKey() == "")
+                && this.config.apiKey == "") {
+            throw new MeiliSearchException(
+                    "An api key is required in the client or should be passed as an argument.");
+        }
+        if (options.getExpiresAt() != null
+                && Date.from(Instant.now()).after((Date) options.getExpiresAt())) {
+            throw new MeiliSearchException("The date expiresAt should be in the future.");
+        }
+        if (searchRules == null) {
+            throw new MeiliSearchException(
+                    "The searchRules field is mandatory and should be defined.");
+        }
+
+        String secret;
+        if (options.getApiKey() == null) {
+            secret = this.config.apiKey;
+        } else {
+            secret = (String) options.getApiKey();
+        }
+
+        // Encrypt the key
+        Algorithm algorithm = Algorithm.HMAC256(secret);
+
+        // Create JWT
+        String jwtToken =
+                JWT.create()
+                        .withClaim("searchRules", searchRules)
+                        .withClaim("apiKeyPrefix", secret.substring(0, 8))
```

Comment:
It is possible to have a `StringIndexOutOfBoundsException` here if this condition **Either starting or ending index is greater than the total number of characters present in the string.** happens

Reply:
Except if the user put the `masterKey` in it it shouldn't happen all the keys are generated by `Meilisearch`.

Reply:
I suggest you add that validation since we cannot prevent the user from adding that key to the method invocation. If it happens, the user will face an exception he does not expects, even before getting the token :)